### DISCUSSION TITLE
Q: How can we avoid hard-coding the html element `lang` attribute?

### DISCUSSION THREAD

**Discussion by Download** [remix-run/remix#6113] [Q&A] [answered]
> To see what I am looking at you can create a new project with `npx create-remix` with `Just the basics` and `Typescript`

Inside `./app/root.tsx` I see:

```tsx
import {
  Links,
  LiveReload,
  Meta,
  Outlet,
  Scripts,
  ScrollRestoration,
} from "@remix-run/react";

export default function App() {
  return (
    <html lang="en">
      <head>
        <meta charSet="utf-8" />
        <meta name="viewport" content="width=device-width,initial-scale=1" />
        <Meta />
        <Links />
      </head>
      <body>
        <Outlet />
        <ScrollRestoration />
        <Scripts />
        <LiveReload />
      </body>
    </html>
  );
}
```

I want to change the `lang` provided here:

```html
    <html lang="en">
```

But instead of hardcoding it to say `"nl"` or `"de"` or whatever, I want this to be parameterized because not all pages in the app will have the same language.

How can I go about this? Is there a way for me to influence how the `Root` component is called/mounted?

What language the page is in will be determined by the URL. On some systems it might be a subdomain (`en.example.com`, `de.example.com`, etc), on others it might be in a path section (`example.com/en`, `example.com/de`, etc) but if we can access the current URL from inside the `Root` component that would also enable us to do it I guess. 

Any help would be greatly appreciated  🙏

**Comment by kiliman** [comment, marked as answer]
Return the desired `lang` from your root loader, then in `App` get it from `useLoaderData`.

```ts
// root.tsx
export async function loader({ request }: LoaderArgs) {
  const lang = await getLanguage(request)
  return json({ lang })
}

export default function App() {
  const { lang } = useLoaderData<typeof loader>()
  return (
    <html lang={lang}>
      //...
    </html>
  )
}
```

**Comment by Download** [comment]
Thanks! I had read about this but for some reason it did not click that I can use this inside the `Root` component just as well.
